lib/migration/finance.rb

```diff
@@ -110,7 +110,7 @@ def self.freeze_closed_invoices
       end
 
       def self.find_invoice_zombies
-        ids = Invoice.connection.execute("select invoices.id from invoices join accounts on invoices.buyer_account_id = accounts.id where invoices.state in ('finalized', 'pending', 'open', 'unpaid') and accounts.deleted_at IS NOT NULL").map(&:first)
+        ids = Invoice.connection.execute("select invoices.id from invoices join accounts on invoices.buyer_account_id = accounts.id where invoices.state in ('finalized', 'pending', 'open', 'unpaid') and accounts.state = 'scheduled_for_deletion'").map(&:first)
```

Comment:
I asked Gui about this specific change of this line and he said:
![image](https://user-images.githubusercontent.com/11318903/49231107-7f1bdb00-f3f1-11e8-8f06-ee6410f2a665.png)